streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala

```diff
@@ -1279,7 +1279,59 @@ class ZStream[-R, +E, +A](val channel: ZChannel[R, Any, Any, Any, E, Chunk[A], A
   final def interleaveWith[R1 <: R, E1 >: E, A1 >: A](
     that: ZStream[R1, E1, A1]
   )(b: ZStream[R1, E1, Boolean]): ZStream[R1, E1, A1] =
-    ???
+    new ZStream(
+      ZChannel.managed {
```

Comment:
Hmm this was previously implemented using `combine`. If we still need to use `toPull` here, could we keep the previous implementation? That means that when (if) we find a way to implement combine/combineChunks without toPull, this implementation would be solved as well.

Reply:
Makes sense, @iravid. I will update.